services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/allu/AlluApplicationData.kt

```diff
@@ -4,41 +4,85 @@ import java.time.ZonedDateTime
 import org.geojson.GeometryCollection
 
 sealed interface AlluApplicationData {
+    val postalAddress: PostalAddress?
     val name: String
+    val customerWithContacts: CustomerWithContacts
+    val representativeWithContacts: CustomerWithContacts?
+    val invoicingCustomer: Customer?
+    val geometry: GeometryCollection
+    val startTime: ZonedDateTime
+    val endTime: ZonedDateTime
     val pendingOnClient: Boolean
+    val identificationNumber: String
+    val customerReference: String?
+    val area: Double?
+    val trafficArrangementImages: List<Int>?
+    val clientApplicationKind: String
 }
 
 data class AlluCableReportApplicationData(
-    // Common, required
+    override val postalAddress: PostalAddress? = null,
     override val name: String,
-    val customerWithContacts: CustomerWithContacts,
-    val geometry: GeometryCollection,
-    val startTime: ZonedDateTime,
-    val endTime: ZonedDateTime,
+    override val customerWithContacts: CustomerWithContacts,
+    override val representativeWithContacts: CustomerWithContacts? = null,
```

Comment:
Shouldn't these be - at least roughly - in the same order as in ApplicationData?

Reply:
Allu classes follow the field order of Allu API description. In my opinion that is more clear. Allu classes reflect Allu and Haitaton classes reflect Haitaton.

Reply:
I think it's more useful to see the difference between Allu and Haitaton data structures easily than to match the current order of the fields in some external sources.

Reply:
Maybe you're right. I will adjust these classes.